mantidproject/mantid
# Display averaged two theta on ALFView

**Description of work.**
This PR adds a QLineEdit to the ALFView interface to display the averaged two theta value of the extracted tubes. It currently uses the two theta of the pixel which is right clicked, but this will be changed in a future revision to make sure the two theta it uses belongs to the peak in the tube.

**To test:**
Load data ALF82301
Click on Pick tab
Expand Rebin section. Filter by d-spacing range 5.5,0.01,6
Click on Select whole tube icon
Right click on the centre yellow square in the instrument view and select Extract Single Tube
There should be a Two theta value on the right hand side of the GUI
Adding a tube to the average should update this two theta value
The tooltip of the Two theta box should show how the average was calculated

Fixes #34568

*This does not require release notes* because **release notes will be added in a later PR**




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
